### PR TITLE
Jsonnet: replace -store.max-query-length with -query-frontend.max-total-query-length

### PR DESCRIPTION
**Pull request by pracucci**
#### What this PR does
In the query-frontend, `-store.max-query-length` is just the fallback for `-query-frontend.max-total-query-length`. `-query-frontend.max-total-query-length` should actually be preferred instead of `-store.max-query-length`, because we're about to deprecate `-store.max-query-length` (in the next release).

**Helm**: the PR https://github.com/grafana/mimir/pull/2998 ignored the diff of `-store.max-query-length` between Helm and Jsonnet, because `-store.max-query-length` was both used in the query-frontend and querier (with different settings) so it conflicted with the idea of having 1 single config file in Helm. However, this is what we're currently fixing (reason why `-store.max-query-length` will be deprecated in the next release), so we can actually introduce `max_total_query_length: 12000h` in Helm in this PR.

#### Which issue(s) this PR fixes or relates to

N/A

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by jhesketh — APPROVED** [review]
lgtm

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 1]
```diff
@@ -277,6 +277,8 @@ mimir:
     {{- end }}
 
     limits:
+      # Limit queries to 500 days. This can be overridden on a per-tenant basis.
```
Comment (osg-grafana) [inline operations/helm/charts/mimir-distributed/values.yaml:280]:
```suggestion
      # Limit queries to 500 days. You can override this on a per-tenant basis.
```

File: operations/mimir/query-frontend.libsonnet [hunk 2]
```diff
@@ -21,7 +21,7 @@
       'query-frontend.max-cache-freshness': '10m',
 
       // Limit queries to 500 days, allow this to be override per-user.
-      'store.max-query-length': '12000h',  // 500 Days
+      'query-frontend.max-total-query-length': '12000h',  // 500 Days
```
Comment (osg-grafana) [inline operations/mimir/query-frontend.libsonnet:24]:
```suggestion
      'query-frontend.max-total-query-length': '12000h',  // 500 days
```

File: operations/mimir/query-frontend.libsonnet [hunk 3]
```diff
@@ -21,7 +21,7 @@
       'query-frontend.max-cache-freshness': '10m',
 
       // Limit queries to 500 days, allow this to be override per-user.
```
Comment (osg-grafana) [inline operations/mimir/query-frontend.libsonnet:23]:
```suggestion
      // Limit queries to 500 days; allow this to be overridden per user.
```

Comment (osg-grafana) [reply operations/mimir/query-frontend.libsonnet:23]: Rationale: matches phrasing of https://github.com/grafana/mimir/pull/3879/files#diff-f600bfee9a33d23b0945bda697930b5d1805149ab3adda8f63458073b3988a1bR280.

**Comment by osg-grafana** [comment]
For possible context for the reader, why was `12000h` chosen and what are its pros and cons? How do I benefit (or not) from this particular length? Is there any gotcha that I need to be aware of as a reader?

(Asking prompting questions that might or might not make sense in this context; feel free to flesh out context as makes sense.)

**Review by osg-grafana — APPROVED** [review]
Unblocking with nits

File: operations/mimir/query-frontend.libsonnet (hunk 3 above)
Comment (osg-grafana) [reply operations/mimir/query-frontend.libsonnet:23]: Just to be 100% clear, "user" means a human here, right?

**Pull request merged**